pypit/arcyarc.pyx

```diff
@@ -438,6 +438,7 @@ def detections_sigma(np.ndarray[DTYPE_t, ndim=1] pixels not None,
             d=1
             flag = 0
             while True:
+                if p+d >= (sz_p-1): break
```

Comment:
I feel like I had it that way first.  But I'm doing as you say as I 
think you should be right.